src/main/java/org/duckdb/DuckDBNative.java

```diff
@@ -15,38 +15,47 @@
 
 final class DuckDBNative {
     static {
-        try {
-            String os_name = "";
-            String os_arch;
-            String os_name_detect = System.getProperty("os.name").toLowerCase().trim();
-            String os_arch_detect = System.getProperty("os.arch").toLowerCase().trim();
-            switch (os_arch_detect) {
-            case "x86_64":
-            case "amd64":
-                os_arch = "amd64";
-                break;
-            case "aarch64":
-            case "arm64":
-                os_arch = "arm64";
-                break;
-            case "i386":
-                os_arch = "i386";
-                break;
-            default:
-                throw new IllegalStateException("Unsupported system architecture");
-            }
-            if (os_name_detect.startsWith("windows")) {
-                os_name = "windows";
-            } else if (os_name_detect.startsWith("mac")) {
-                os_name = "osx";
-                os_arch = "universal";
-            } else if (os_name_detect.startsWith("linux")) {
-                os_name = "linux";
-            }
-            String lib_res_name = "/libduckdb_java.so"
-                                  + "_" + os_name + "_" + os_arch;
+        String libname = "duckdb_java";
+        boolean loaded = loadLibraryByName(libname);
```

Comment:
- as far as I can say, the PR:
   - asks the JVM to load the library by name first
   - only if that fails, it "tries harder" to find the library itself
- using `System.loadLibrary` is [runtime container friendly](https://wiki.apidesign.org/wiki/NetBeans_Runtime_Container) and packagers of full applications (like [Enso](http://github.com/enso-org/enso)) friendly
- "trying harder" is _"Maven friendly"_ - enough to add dependency to `pom.xml` and things work